Add/update notebooks

klimaj opened the pull request:
This PR adds and updates Jupyter notebook examples and is ready for review. If errors arise on the testing server I will plan to open a new PR addressing them.

Additional tasks:

- [x] Add notice about using `jupyterlab` instead of `jupyter notebooks` for notebooks using `pyrosetta.distributed.viewer`
- [x] Add notice about installing third-party external packages in `01.00-How-to-Get-Started.ipynb`: `matplotlib biopython blosc dask distributed jupyter numpy pandas py3Dmol scipy traitlets graphviz seaborn dask-jobqueue`

klimaj commented:
Hi @lyskov, this PR is ready for review. Upon merging this into the `PyRosetta.notebooks` submodule I will update the submodule HEAD from #4279 and queue testing of the notebooks (which now include ipython magic commands and third-party package imports) from #4279. If any errors arise on the testing server from this PR, I can open a new PR addressing them.

jadolfbr commented on notebooks/01.02-PyRosetta-Google-Drive-Usage-Example.ipynb:
```diff
@@ -42,11 +42,12 @@
+    "    !pip install pyrosettacolabsetup\n",
+    "    import pyrosettacolabsetup\n",
+    "    pyrosettacolabsetup.setup()\n",
+    "    print (\"Notebook is set for PyRosetta use in Colab.  Have fun!\")"
```
Why not print this in the setup function?

klimaj replied:
Agreed! That would be cool. Here I just re-implemented the setup functionality that was already implemented in other notebooks, just wrapped in an `if 'google.colab' in sys.modules:`.

jadolfbr commented:
GNU-Parellel
=========

This is also a great notebook.  Some comments that need to be addressed.

1) You have the user cat expected_outputs/run_file_parallel_interactive.txt  instead of the one specific for their work environment in output that you had created in the block before. I would also use the command `parallel` instead of the full path as the path can be different (and certainly is on my mac).  If it's not found - instruct the user how to install it on linux/mac. Or just point to the GNU-parellel website for more info.

2) Users may not know what SLURM is.   I would include a description of what slurm is. Explain that this is for a cluster-environment, and that SLURM is just one of several job-submission applications and that their cluster may not have SLURM.  Also note that the notebook is setup from via hosting in a cluster environment.  Provide links for how to set this up.

dask-delayed
==========

1) I would change the name of this to Dask-Delayed via SLURM or something like that. 

2) I could not run this locally. I would put a disclaimer on the top of this notebook that it must be run on a SLURM-capable machine, most likely a cluster. 

dask-delayed-PyRosetta
==================

1) I would combine this with dask-delayed.  I would have it in a second major section. 

2) Again, I could not run this locally. 

3) Make sure users have graphviz installed to see the visualization you have here. 

ligand-docking-Dask
===============

1) Same for above.  Combine the completed and main notebooks to one notebook.  If there is a way to have the client not be slurm, please add these. 


Thanks for writing these Jason!  I think these are really cool and a great addition.  Cheers!

klimaj commented:
@jadolfbr many thanks for your review! I have now updated the notebooks addressing your comments. The only point that I could not directly address in the notebooks is that the interactive slider widget did not work for you, which requires the `ipywidgets` module so you might try `pip install jupyter` to fix that. I have also added third-party dependency installation instructions to 01.00 and 01.01. I think this PR is ready to merge pending your review. Thank you again!
